kubernetes/samples/scripts/common/utility.sh

```diff
@@ -447,6 +447,8 @@ function createFiles {
     sed -i -e "s:%CUSTOM_RCUPREFIX%:${rcuSchemaPrefix}:g" ${createJobOutput}
     sed -i -e "s|%CUSTOM_CONNECTION_STRING%|${rcuDatabaseURL}|g" ${createJobOutput}
     sed -i -e "s:%EXPOSE_T3_CHANNEL_PREFIX%:${exposeAdminT3Channel}:g" ${createJobOutput}
+    sed -i -e "s:%FRONTEND_HOST%:${frontEndHost}:g" ${createJobOutput}
+    sed -i -e "s:%FRONTEND_PORT%:${frontEndPort}:g" ${createJobOutput}
```

Comment:
(3) Is the need to set these properties common? If so, the corresponding inputs file for the sample should be updated to include 'frontEndHost' and 'frontEndPort' (commented out, with a comment describing their purpose).

Reply:
(1) 

> (1) For this to work, I assume there needs to be a corresponding change in `create-domain-job-template.yaml` that references the `%FRONTEND_HOST%` and `%FRONT_ENDPORT%` macros? If the answer is yes. the change is missing from this pull.

This is part of IDM release. We are providing our samples and templates which include these properties.

Reply:
> (2) Are these properties for a narrow use case? If so, can you use macro/setting names that's are more specific/descriptive than FRONTEND_HOST/FRONTEND_PORT/frontEndHost/frontEndPort?

These are the standard documented properties used by IDM product

Reply:
> (3) Is the need to set these properties common? If so, the corresponding inputs file for the sample should be updated to include 'frontEndHost' and 'frontEndPort' (commented out, with a comment describing their purpose).

Similar to point 1 input files will be shipped when idm samples will be pushed to the repo

Reply:
> (1)
> 
> > (1) For this to work, I assume there needs to be a corresponding change in `create-domain-job-template.yaml` that references the `%FRONTEND_HOST%` and `%FRONT_ENDPORT%` macros? If the answer is yes. the change is missing from this pull.
> 
> This is part of IDM release. We are providing our samples and templates which include these properties.

(1A) Oh. The point still stands - the macro names should be less specific.

(1B) But, since this is for an external sample/product, I think this brings the entire pull into question.  IMO, external samples and products probably should not directly rely on WKO sample shell scripts and templates. I think the WKO samples and their internals aren't given the same 'backward compatibility' guarantees as WKO's public APIs.  They are not an API per se.   @rjeberhard @markxnelson Thoughts?

(1C) If it turns out the pull's use case is still in play even after considering 1B, then I think a different approach may be advisable that supports a more custom 'future' proof method for handling arbitrary inputs from external use cases that want to fine tune a sample.  For example, we could document support for any input in the input file that begins with '_custom_', and utility.sh could find every environment variable that starts with '_custom_' and substitute the corresponding value in its target template(s).  Something like: 
`for env_var in $(env | grep '^_custom_'); do  sed -i -e "s:%${env_var}%:${!env_var}:g" ${createJobOutput} ; done`  
(The '${!' syntax provides the value for given macro name.)

Reply:
> > (1)
> > > (1) For this to work, I assume there needs to be a corresponding change in `create-domain-job-template.yaml` that references the `%FRONTEND_HOST%` and `%FRONT_ENDPORT%` macros? If the answer is yes. the change is missing from this pull.
> > 
> > 
> > This is part of IDM release. We are providing our samples and templates which include these properties.
> 
> (1A) Oh. The point still stands - the macro names should be less^H^H^H^H more specific.
> 
> (1B) But, since this is for an external sample/product, I think this brings the entire pull into question. IMO, external samples and products probably should not directly rely on WKO sample shell scripts and templates. I think the WKO samples and their internals aren't given the same 'backward compatibility' guarantees as WKO's public APIs. They are not an API per se. @rjeberhard @markxnelson Thoughts?
> 
> (1C) If it turns out the pull's use case is still in play even after considering 1B, then I think a different approach may be advisable that supports a more custom 'future' proof method for handling arbitrary inputs from external use cases that want to fine tune a sample. For example, we could document support for any input in the input file that begins with '_custom_', and utility.sh could find every environment variable that starts with '_custom_' and substitute the corresponding value in its target template(s). Something like:
> `for env_var in $(env | grep '^_custom_'); do sed -i -e "s:%${env_var}%:${!env_var}:g" ${createJobOutput} ; done`
> (The '${!' syntax provides the value for a given macro name.)

What is the current recommendation if any new property needs to be added?

Reply:
We can make the macro more specific to our product. We see properties already added for FMW infra domains under following comment.
 # extra entries for FMW Infra domains

Also, in my opinion, point 1C will solve more of our future use cases as well